front/src/App.tsx

```diff
@@ -6,22 +6,47 @@ import SpellbookContainer, {
 } from "components/spellbookContainer/SpellbookContainer";
 import { Character } from "types";
 import { ClassName } from "common/character";
+import useStateWithLocalStorage from "hooks/useStateWithLocalStorage";
 
 const INITIAL_CHARACTER: Character = {
     name: "Josh Mann",
     class: ClassName.Sorcerer,
     spellCastingModifier: 0,
     showSpellSaveDC: true,
+    id: "",
 };
 
+const CHARACTERS_KEY = "characters";
+
 function App() {
-    const [character, setCharacter] = useState<Character>(INITIAL_CHARACTER);
+    const fallbackCharacter: Character = {
+        ...INITIAL_CHARACTER,
+        id: crypto.randomUUID(),
+    };
+
+    const [characters, setCharacters] = useStateWithLocalStorage<Character[]>(
+        CHARACTERS_KEY,
+        [fallbackCharacter]
+    );
+    const [currentCharacterID] = useState<string>(characters[0].id);
```

Comment:
We should probably set the `currentCharacterId` into local stoage as well, that way it persists across sessions.
```suggestion
    const [currentCharacterId, setCurrentCharacterId] = useStateWithLocalStorage<string>(
        CURRENT_CHARACTER_ID_KEY, characters[0].id
    );
```